core/src/main/java/org/infinispan/metrics/impl/AbstractMetricsRegistration.java

```diff
@@ -120,7 +123,12 @@ private Set<Object> registerMetrics(Object instance, String jmxObjectName, Colle
          }
          metricPrefix += NameUtils.decamelize(jmxObjectName) + '_';
       }
-      return internalRegisterMetrics(instance, metrics, metricPrefix);
+      Set<Object> metricIds = new HashSet<>();
+      if (globalConfig.metrics().legacy()) {
+         metricIds.addAll(internalRegisterMetrics(instance, metrics, VENDOR_PREFIX + legacyNamePrefix + metricPrefix));
+      }
+      metricIds.addAll(internalRegisterMetrics(instance, metrics, INFINISPAN_PREFIX + metricPrefix));
```

Comment:
@cfredri4 here is the problem, it needs to be exclusive. The micrometer "gives" Infinispan counters and timer objects with which Infinispan interacts. The first registration objects are not used as Infinispan only interacts with the second registration objects.

Reply:
Thanks @pruivo, that explains it.
I will change to not double-register the old and new metrics and instead only register the new metrics if `legacy=false`.